doc/design/zarr-manifests.md

```diff
@@ -0,0 +1,150 @@
+Zarr Manifest Files
+===================
+
+This document specifies *Zarr manifest files*, each of which describes a Zarr
+in the Dandi Archive, including the Zarr's internal directory structure and
+details on all of the Zarr's *entries* (regular, non-directory files).  The
+Dandi Archive is to automatically generate these files and serve them via S3.
+
+@yarikoptic has already produced proof-of-concept manifest files for all Zarrs
+in the Dandi Archive at <https://github.com/dandi/zarr-manifests>.  Except
+where noted, the manifest file format defined herein matches the format used by
+the proof of concept.
+
+
+Archive Behavior
+----------------
+
+Whenever Dandi Archive calculates the checksum for a Zarr in the Archive, it
+shall additionally produce a *manifest file* listing various information about
+the Zarr and its entries in the format described in the next section.  This
+manifest file shall be stored in the Archive's S3 bucket at the path
+`zarr-manifest/{zarr_id}.json`, where `{zarr_id}` is replaced by the ID of the
```

Comment:
here we indeed should introduce versioning similar (if not identical) to the one currently present in https://github.com/dandi/zarr-manifests/tree/master/zarr-manifests-v2-sorted .

Reply:
Done: 40ebed86